### PR TITLE
[New package] Add OpenCC v1.1.3

### PR DESCRIPTION
**Pull request by yqwu905**
I add a package [OpenCC](https://github.com/BYVoid/OpenCC) for convert between Traditional and Simplified Chinese.
Due to some unknown issue, build only work for linux x86_64 now.

### CODE REVIEW

File: O/OpenCC/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+using BinaryBuilder
+
+name = "libopencc"
```
Comment (giordano) [inline O/OpenCC/build_tarballs.jl:3]: This name ***must*** match the name of the directory, so you have to either change this name or rename the directory.  I'd suggest doing the former
```suggestion
name = "OpenCC"
```

File: O/OpenCC/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+version = v"1.1.3"
+sources = [
+  ArchiveSource("https://github.com/BYVoid/OpenCC/archive/refs/tags/ver.1.1.3.tar.gz", "99a9af883b304f11f3b0f6df30d9fb4161f15b848803f9ff9c65a96d59ce877f")
```
Comment (giordano) [inline O/OpenCC/build_tarballs.jl:6]:
```suggestion
  ArchiveSource("https://github.com/BYVoid/OpenCC/archive/refs/tags/ver.$(version).tar.gz", "99a9af883b304f11f3b0f6df30d9fb4161f15b848803f9ff9c65a96d59ce877f")
```

File: O/OpenCC/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/OpenCC-ver.1.1.3
```
Comment (giordano) [inline O/OpenCC/build_tarballs.jl:10]:
```suggestion
cd ${WORKSPACE}/srcdir/OpenCC*
```

File: O/OpenCC/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+cmake -S. -Bbuild -DCMAKE_INSTALL_PREFIX:PATH=${prefix}
+cmake --build build --config Release --target install
+"""
+
+platforms = [
+  Platform("x86_64", "linux"),
+]
+
```
Comment (giordano) [inline O/OpenCC/build_tarballs.jl:18]: From the log:
```
┌ Warning: /agent/_work/1/s/O/OpenCC/build/x86_64-linux-gnu/q4bCPTex/x86_64-linux-gnu-libgfortran5-cxx11/destdir/bin/opencc_phrase_extract contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/gIX21/src/auditor/compiler_abi.jl:247
```
```suggestion
platforms = expand_cxxstring_abis(platforms)

```

File: O/OpenCC/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+products = [
+  LibraryProduct("libopencc", :libopencc),
+  FileProduct("share/opencc/TSCharacters.ocd2", :opencc_char),
+  FileProduct("share/opencc/TSPhrases.ocd2", :opencc_phra),
+  FileProduct("share/opencc/t2s.json", :opencc_conf)
+]
+
+dependcies = []
```
Comment (giordano) [inline O/OpenCC/build_tarballs.jl:26]:
```suggestion
dependcies = Dependency[]
```

File: O/OpenCC/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependcies, preferred_gcc_version = v"11.1.0")
```
Comment (giordano) [inline O/OpenCC/build_tarballs.jl:28]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependcies; julia_compat="1.6")
```

File: O/OpenCC/build_tarballs.jl [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+name = "libopencc"
+version = v"1.1.3"
+sources = [
+  ArchiveSource("https://github.com/BYVoid/OpenCC/archive/refs/tags/ver.1.1.3.tar.gz", "99a9af883b304f11f3b0f6df30d9fb4161f15b848803f9ff9c65a96d59ce877f")
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/OpenCC-ver.1.1.3
+cmake -S. -Bbuild -DCMAKE_INSTALL_PREFIX:PATH=${prefix}
+cmake --build build --config Release --target install
```
Comment (giordano) [inline O/OpenCC/build_tarballs.jl:12]: Please read the [documentation](https://docs.binarybuilder.org/stable/build_tips/#CMake-builds).  In particular, note that setting `CMAKE_TOOLCHAIN_FILE` is ***necessary*** to actually use the right compiler
```suggestion
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    ..
make -j${nproc}
make install
```

**Comment by yqwu905** [comment]
Thanks for your instructions!
I just submit a commit to this pr, I fix problems you point out, except:
* I manually set `cxxstring_abi="cxx11"` for platform, instead of use `expand_cxxstring_abis`

Also, I delete some products and specification of gcc compiler.

**Comment by giordano** [comment]
> fix problems you point out, except:
> 
>     * I manually set `cxxstring_abi="cxx11"` for platform, instead of use `expand_cxxstring_abis`

Why??

You ignored my comments about setting `CMAKE_TOOLCHAIN_FILE` and `julia_compat="1.6"`.

**Pull request closed without merging**